odoo/models.py

```diff
@@ -4675,13 +4675,13 @@ def exists(self):
 
         By convention, new records are returned as existing.
         """
-        ids, new_ids = [], []
-        for i in self._ids:
-            (new_ids if isinstance(i, NewId) else ids).append(i)
+        new_ids, ids = partition(lambda i: isinstance(i, NewId), self._ids)
         if not ids:
             return self
-        query = """SELECT id FROM "%s" WHERE id IN %%s""" % self._table
-        self._cr.execute(query, [tuple(ids)])
+        query = Query(self.env.cr, self._table, self._table_query)
+        query.add_where(f'"{self._table}".id IN %s')
+        query_str, params = query.select()
+        self.env.cr.execute(query_str, params + [tuple(ids)])
```

Comment:
I suggest to add the parameters in the where clause
```suggestion
        query = Query(self.env.cr, self._table, self._table_query)
        query.add_where(f'"{self._table}".id IN %s', [tuple(ids)])
        query_str, params = query.select()
        self.env.cr.execute(query_str, params)
```